edn/src/edn.rustpeg

```diff
@@ -105,37 +104,41 @@ keyword -> Value
     }
 
 #[export]
-list -> Value = "(" __ v:(__ value)* __ ")" {
+list -> Value = "(" v:(value)* ")" {
     Value::List(LinkedList::from_iter(v))
 }
 
 #[export]
-vector -> Value = "[" __ v:(__ value)* __ "]" {
+vector -> Value = "[" v:(value)* "]" {
     Value::Vector(v)
 }
 
 #[export]
-set -> Value = "#{" __ v:(__ value)* __ "}" {
+set -> Value = "#{" v:(value)* "}" {
     Value::Set(BTreeSet::from_iter(v))
 }
 
-pair -> (Value, Value) = k:(value) " " v:(value) ", "? {
+pair -> (Value, Value) = k:(value) v:(value) {
     (k, v)
 }
 
 #[export]
-map -> Value = "{" __ v:(pair)* __ "}" {
+map -> Value = "{" v:(pair)* "}" {
     Value::Map(BTreeMap::from_iter(v))
 }
 
 // It's important that float comes before integer or the parser assumes that
 // floats are integers and fails to parse
 #[export]
 value -> Value
-    = nil / boolean / float / bigint / integer / text /
+    = __ v:(nil / boolean / float / bigint / integer / text /
       keyword / symbol /
-      list / vector / map / set
+      list / vector / map / set) __ {
+    v
+}
+
+whitespace = (" " / "\r" / "\n" / "\t" / ",")
```

Comment:
It's worth commenting that Clojure, and thus EDN, regard commas as whitespace, and thus this is not a mistake; `[5,,,,4]` is totally legit syntax for a two-element array.

(It's also worth noting that idiomatic EDN doesn't bother with commas.)